### PR TITLE
20231008 remove expect used

### PR DESCRIPTION
**Pull request by Firstyear**
Removes a ton of expect used. Note clippy is *not* run yet, because I added some unwraps that I plan to get rid of soon.

Checklist

- [ x ] This pr contains no AI generated code
- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: server/core/src/lib.rs [hunk 1]
```diff
@@ -173,7 +173,13 @@ macro_rules! dbscan_setup_be {
 
 pub fn dbscan_list_indexes_core(config: &Configuration) {
```
Comment (yaleman) [inline server/core/src/lib.rs:174]: can we add an explicit return type here? 🤔

File: server/lib/src/filter.rs [hunk 2]
```diff
@@ -1287,8 +1287,7 @@ impl FilterResolved {
 
                 // If the f_list_or only has one element, pop it and return.
                 if f_list_new.len() == 1 {
-                    #[allow(clippy::expect_used)]
-                    f_list_new.pop().expect("corrupt?")
+                    f_list_new.remove(0)
```
Comment (yaleman) [inline server/lib/src/filter.rs:1290]: This should also be a [clear](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.clear)
```suggestion
                    f_list_new.clear()
```

**Review by yaleman — CHANGES_REQUESTED** [review]
Minor changes for clarity but woo, less expecting things.

File: server/lib/src/filter.rs (hunk 2 above)
Comment (Firstyear) [reply server/lib/src/filter.rs:1290]: Uhhh? no? clear just deletes everythdng. We want to *pop* the last element and return it.

Comment (yaleman) [reply server/lib/src/filter.rs:1290]: Derp I r dumb. All good 😄

**Review by yaleman — APPROVED** [review]

**Pull request merged**